docs/adrs/multus.md

```diff
@@ -0,0 +1,36 @@
+# Record architecture decisions
+
+Date: 2024-02-07
+
+## Status
+
+Discussing
+
+## Context
+
+### Multus
+
+Multus is a CNI multiplexer that allows pods to have multiple network interfaces. We have users that are operating K3s + Multus but it is not super obvious how to configure it to work with K3s and how to add the additional pieces needed (e.g. IPAM or additional CNI plugins). We could facilitate this by creating an integration with Multus. Moreover, Multus deployments normally come with a CNI IPAM plugin to distribute IPs to the extra interfaces. The most popular one among Multus users is Whereabouts, so we will take the decision to use that one for this integration.
+
+### Design suggestion
+
+The suggestion is to include a new `--multus` boolean flag. When that flag gets activated, we deploy a Kubernetes job that will run in that node and will:
+* Run the [cni-plugins image](https://github.com/rancher/image-build-cni-plugins) which will deploy in the node all the widely used CNI binaries (e.g. macvlan), the multus binary and the whereabouts binary
+* Install the required multus CRD `network-attachment-definitions.k8s.cni.cncf.io` and the whereabouts CRDs `ippools.whereabouts.cni.cncf.io` & `overlappingrangeipreservations.whereabouts.cni.cncf.io`
+* Create the appropriate Multus CNI config file and place it in the right path
+* Start the whereabouts controller
+
+These tasks are more or less what the upstream Multus and whereabouts daemonset do but, at least for Multus, it seems too overkill to have a daemonset running forever just for deploying the binaries and creating the correct multus cni config. A job that runs the `cni-plugins image` once the nodes comes up seems like a better solution.
```

Comment:
I'm not sure a Job is the best way to do this, given that that the process for assigning job pods to specific nodes is a bit clumsy. We would need to reinvent much of the logic around creating and managing job-node assignment, and this logic already exists in the daemonset controller. I think we should either drop the required files directly from the k3s supervisor process, or use a daemonset to ensure that the install pod gets run on every node.

Reply:
if it is too clumsy, then perhaps it is best for k3s to do that job but then, I don't want to increase the size of k3s by adding too many binaries that most people won't use... I'm starting to wonder if perhaps the idea of that user that suggested to add "macvlan" to the cni multi-exec might be the best one :D

Reply:
macvlan + multus + whereabouts +.... I'll have fun in the next days

Reply:
Adding multus+whereabouts to the multi-exec is not worth it.:
* Multus binary increases the size of k3s significantly. Given that only a handful of users will benefit from this, I don't think it makes sense
* whereabouts binary has very very old dependencies. That project actually looks sort of unmaintained. Anyway, it would creep in a lot of CVEs and we don't want that

Reply:
Open question: do we want to add the multus images to the core k3s airgap image list, or should we have an optional image tarball, like we do for rke2?

Reply:
Good point, probably the latter makes more sense

Reply:
Disadvantage is that we have to update build scripts, QA processes, and documentation to handle the optional tarballs. We have never had more than 1 for k3s.

Reply:
yeah. To minimize that pain, I'd leave the same name for the current tarball k3s-airgap-images-${ARCH}.tar and the images file `k3s-images.txt`